### PR TITLE
Fixed Windows build issues

### PR DESCRIPTION
**Pull request by chrismidgley**
Fixed issue with building on WIndows under Atom or VS Code.  Both use PowerShell, which doesn't work correctly with CMD files, and requires backslash instead of forward.  Also improved error handling to help debug problems.

### CODE REVIEW

File: .build_web.py [hunk 1]
```diff
@@ -20,14 +20,25 @@ def build_web():
         os.chdir("web")
         print("Attempting to build webpage...")
         try:
-            print check_output(["npm", "install"])
-            print check_output(["node_modules/.bin/gulp"])
+            if platform.system() == "Windows":
+                print check_output(["npm.cmd", "install", "--only=dev"])
+                print check_output(["node_modules\\.bin\\gulp.cmd"])
+            else:
+                print check_output(["npm.cmd", "install"])
```
Comment (sidoh) [inline .build_web.py]: should this be `npm` instead of `npm.cmd`?

File: .build_web.py [hunk 2]
```diff
@@ -20,14 +20,25 @@ def build_web():
...
+                print check_output(["node_modules/.bin/gulp"])
             copyfile("build/index.html.gz.h", "../dist/index.html.gz.h")
-
+        except OSError as e:
+            print "Encountered error OSError building webpage: ", e
+            if e.filename:
+                print "Filename is", e.filename
```
Comment (sidoh) [inline .build_web.py]: not that it matters much, but should there be a space at the end of the string here?

**Comment by chrismidgley** [comment]
Crud.  Yes, the non-Windows leg should be "npm" not "npm.cmd".  My bad, as I can't test it (right now at least) on Linux.  

The space was there before and I left it, but I'll remove it as well.

Update coming in a bit.

**Comment by sidoh** [comment]
The Travis build should run automatically when you push commits to a PR.  I was just checking the build logs here:

https://travis-ci.org/sidoh/esp8266_milight_hub/builds/341688382

**Comment by chrismidgley** [comment]
Pulling - will republish under "cmidgley" account

**Pull request closed without merging**